engine/src/main/scala/cromwell/engine/workflow/package.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell.engine
+
+import akka.actor.{ActorSystem, ActorRef}
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.services.MetadataServiceActor.{MetadataServiceAction, PutMetadataAction, MetadataServiceResponse, MetadataEvent}
+
+import scala.concurrent.{Future, ExecutionContext}
+
+package object workflow {
+  // Sent when there is a failure to push to the metadata service
+  private[workflow] case class MetadataServiceFailureResponse(reason: Throwable, action: Option[MetadataServiceAction] = None)
+
+  object WorkflowMetadataKeys {
+    val Name = "workflowName"
+    val Id = "id"
+    val SubmissionTime = "submission"
+    val StartTime = "start"
+    val EndTime = "end"
+    val Status = "status"
+    val Outputs = "outputs"
+    val Inputs = "inputs"
+    val Calls = "calls"
+  }
+
+  implicit class EnhancedServiceRegistry(val serviceRegistryActor: ActorRef) extends StrictLogging {
+    def publishToMetadataService(message: MetadataEvent)(implicit actorSystem: ActorSystem, ec: ExecutionContext): Future[MetadataServiceResponse] = {
```

Comment:
I'm really not a fan of using enhanced classes to send messages to actors like this.